index.d.ts

```diff
@@ -1059,6 +1084,88 @@ declare namespace tracer {
     eventTrackingV2: EventTrackingV2
   }
 
+  /**
+   * Feature Flagging Provider (OpenFeature-compatible).
+   *
+   * Wraps @datadog/openfeature-node-server with Remote Config integration for dynamic flag configuration.
+   * Implements the OpenFeature Provider interface for flag evaluation.
+   *
+   * @beta This feature is in preview and not ready for production use
+   */
+  export interface FlaggingProvider {
+    /**
+     * Metadata about this provider.
+     */
+    metadata: { name: string };
+
+    /**
+     * Resolves a boolean flag value.
+     *
+     * @param flagKey The key of the flag to evaluate
+     * @param defaultValue The default value to return if evaluation fails
+     * @param context Evaluation context (e.g., user attributes)
+     * @param logger Optional logger instance
+     * @returns Promise resolving to evaluation result with value and reason
+     */
+    resolveBooleanEvaluation(flagKey: string, defaultValue: boolean, context?: any, logger?: any): Promise<{ value: boolean; reason: string }>;
+
+    /**
+     * Resolves a string flag value.
+     *
+     * @param flagKey The key of the flag to evaluate
+     * @param defaultValue The default value to return if evaluation fails
+     * @param context Evaluation context (e.g., user attributes)
+     * @param logger Optional logger instance
+     * @returns Promise resolving to evaluation result with value and reason
+     */
+    resolveStringEvaluation(flagKey: string, defaultValue: string, context?: any, logger?: any): Promise<{ value: string; reason: string }>;
+
+    /**
+     * Resolves a number flag value.
+     *
+     * @param flagKey The key of the flag to evaluate
+     * @param defaultValue The default value to return if evaluation fails
+     * @param context Evaluation context (e.g., user attributes)
+     * @param logger Optional logger instance
+     * @returns Promise resolving to evaluation result with value and reason
+     */
+    resolveNumberEvaluation(flagKey: string, defaultValue: number, context?: any, logger?: any): Promise<{ value: number; reason: string }>;
+
+    /**
+     * Resolves an object flag value.
+     *
+     * @param flagKey The key of the flag to evaluate
+     * @param defaultValue The default value to return if evaluation fails
+     * @param context Evaluation context (e.g., user attributes)
+     * @param logger Optional logger instance
+     * @returns Promise resolving to evaluation result with value and reason
+     */
+    resolveObjectEvaluation(flagKey: string, defaultValue: any, context?: any, logger?: any): Promise<{ value: any; reason: string }>;
+
+    /**
+     * Gets the current flag configuration.
+     *
+     * @returns The current configuration object
+     */
+    getConfiguration(): any;
```

Comment:
Using an `any` here could cause issues for consumers with strict type analysis checks.

You could do this at the top of the file:
```
import { DatadogNodeServerProvider } from '@datadog/openfeature-node-server';
type FlaggingConfiguration = ReturnType<DatadogNodeServerProvider['getConfiguration']>;
```

Then, you'd have `getConfiguration(): FlaggingConfiguration`, and `setConfiguration(config: FlaggingConfiguration)`